matrix_sdk_base/src/models/room.rs

```diff
@@ -491,21 +505,28 @@ impl Room {
     /// # Arguments
     ///
     /// * `event` - The event of the room.
-    pub fn receive_timeline_event(&mut self, event: &RoomEvent) -> bool {
-        match event {
-            // update to the current members of the room
-            RoomEvent::RoomMember(member) => self.handle_membership(member),
-            // finds all events related to the name of the room for later use
-            RoomEvent::RoomName(name) => self.handle_room_name(name),
-            RoomEvent::RoomCanonicalAlias(c_alias) => self.handle_canonical(c_alias),
-            RoomEvent::RoomAliases(alias) => self.handle_room_aliases(alias),
-            // power levels of the room members
-            RoomEvent::RoomPowerLevels(power) => self.handle_power_level(power),
-            RoomEvent::RoomTombstone(tomb) => self.handle_tombstone(tomb),
-            RoomEvent::RoomEncryption(encrypt) => self.handle_encryption_event(encrypt),
-            #[cfg(feature = "messages")]
-            RoomEvent::RoomMessage(msg) => self.handle_message(msg),
-            _ => false,
+    pub fn receive_timeline_event(&mut self, event: &AnyRoomEventStub, room_id: &RoomId) -> bool {
+        match &event {
+            AnyRoomEventStub::State(event) => match &event {
+                // update to the current members of the room
+                AnyStateEventStub::RoomMember(event) => self.handle_membership(&event, room_id),
+                // finds all events related to the name of the room for later use
+                AnyStateEventStub::RoomName(event) => self.handle_room_name(&event),
+                AnyStateEventStub::RoomCanonicalAlias(event) => self.handle_canonical(&event),
+                AnyStateEventStub::RoomAliases(event) => self.handle_room_aliases(&event),
+                // power levels of the room members
+                AnyStateEventStub::RoomPowerLevels(event) => self.handle_power_level(&event),
+                AnyStateEventStub::RoomTombstone(event) => self.handle_tombstone(&event),
+                AnyStateEventStub::RoomEncryption(event) => self.handle_encryption_event(&event),
+                _ => false,
+            },
+            AnyRoomEventStub::Message(event) => match &event {
+                #[cfg(feature = "messages")]
+                AnyMessageEventStub::RoomMessage(event) => self.handle_message(&event),
+                // TODO if a redaction event deletes one of our saved messages delete it?
```

Comment:
Probably better to modify it to look like a redacted event, that is remove the content otherwise you'll end  up with a gap instead of showing a redacted event when you restore the messages.

Reply:
We would have to turn the MessageWrapper thing that holds MessageEventStub<MessageEventContent> to an enum to hold the messages and `RedactionEventStub`, they are separate types unfortunatly, they are both AnyMessageEvntStub that would work maybe.